File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/HostedCheckoutMapper.cs
using Business.DTOs.CreateHostedCheckouts;
using Business.DTOs.GetPaymentByHostedCheckoutId;
using Infrastructure.Extensions.Models;
using OnlinePayments.Sdk.Domain;

namespace Infrastructure.Mappers;

public static class HostedCheckoutMapper
{
    public static CreateHostedCheckoutRequest Map(CreateHostedCheckoutRequestDto requestDto)
    {
        return new()
        {
            Order = new()
            {
                AmountOfMoney = new()
                {
                    Amount = (long?)requestDto.Amount,
                    CurrencyCode = requestDto.Currency.ToString()
                },
                Customer = new()
                {
                    PersonalInformation = new()
                    {
                        Name = new()
                        {
                            FirstName = requestDto.BillingAddress?.FirstName,
                            Surname = requestDto.BillingAddress?.LastName
                        }
                    },
                    BillingAddress = new()
                    {
                        City = requestDto.BillingAddress?.City,
                        CountryCode = requestDto.BillingAddress?.Country.ToIsoAlpha2(),
                        Street = requestDto.BillingAddress?.Street,
                        Zip = requestDto.BillingAddress?.Zip
                    }
                },
                Shipping = new()
                {
                    Address = new()
                    {
                        City = requestDto.ShippingAddress?.City,
                        CountryCode = requestDto.ShippingAddress?.Country.ToIsoAlpha2(),
                        Street = requestDto.ShippingAddress?.Street,
                        Zip = requestDto.ShippingAddress?.Zip,
                        Name = new()
                        {
                            FirstName = requestDto.ShippingAddress?.FirstName,
                            Surname = requestDto.ShippingAddress?.LastName
                        }
                    }
                }
            },
            HostedCheckoutSpecificInput = new()
            {
                ReturnUrl = requestDto.RedirectUrl,
                Locale = requestDto.Language.ToLocale(requestDto.BillingAddress?.Country)
            }
        };
    }

    public static CreateHostedCheckoutResponseDto Map(CreateHostedCheckoutResponse? response)
    {
        return new()
        {
            HostedCheckoutId = response?.HostedCheckoutId,
            RedirectUrl = response?.RedirectUrl,
            ReturnMAC = response?.RETURNMAC
        };
    }
    
    public static GetPaymentByHostedCheckoutIdResponseDto Map(GetHostedCheckoutResponse? response)
    {
        return new()
        {
            Status = response?.Status,
            PaymentId = response?.CreatedPaymentOutput?.Payment?.Id
        };
    }
}

File: server/nodejs/src/infrastructure/mappers/hosted-checkout-mapper.ts
import { Domain } from 'onlinepayments-sdk-nodejs';
import { CreateHostedCheckoutRequestDto } from '../../business/dtos/hosted-checkout/create-hosted-checkout-request-dto';
import { CreateHostedCheckoutResponseDto } from '../../business/dtos/hosted-checkout/create-hosted-checkout-response-dto';
import { GetPaymentByHostedCheckoutResponseDto } from '../../business/dtos/hosted-checkout/get-payment-by-hosted-checkout-response-dto';
import { LanguageUtil } from '../../business/utils/models/language-util';
import { CountryUtil } from '../../business/utils/models/country-util';

export const HostedCheckoutMapper = {
    toSdkRequest(requestDto: CreateHostedCheckoutRequestDto): Domain.CreateHostedCheckoutRequest {
        return {
            order: {
                amountOfMoney: {
                    amount: requestDto.amount,
                    currencyCode: requestDto.currency,
                },
                customer: {
                    personalInformation: {
                        name: {
                            firstName: requestDto.billingAddress?.firstName,
                            surname: requestDto.billingAddress?.lastName,
                        },
                    },
                    billingAddress: {
                        city: requestDto.billingAddress?.city,
                        countryCode: CountryUtil.toIsoAlpha2(requestDto.shippingAddress?.country),
                        street: requestDto.billingAddress?.street,
                        zip: requestDto.billingAddress?.zip,
                    },
                },
                shipping: {
                    address: {
                        city: requestDto.shippingAddress?.city,
                        countryCode: CountryUtil.toIsoAlpha2(requestDto.shippingAddress?.country),
                        street: requestDto.shippingAddress?.street,
                        zip: requestDto.shippingAddress?.zip,
                        name: {
                            firstName: requestDto.shippingAddress?.firstName,
                            surname: requestDto.shippingAddress?.lastName,
                        },
                    },
                },
            },
            hostedCheckoutSpecificInput: {
                returnUrl: requestDto.redirectUrl,
                locale: LanguageUtil.toLocale(requestDto.language, requestDto.billingAddress?.country),
            },
        };
    },

    fromSdkResponse(sdkResponse: Domain.CreateHostedCheckoutResponse): CreateHostedCheckoutResponseDto {
        return {
            hostedCheckoutId: sdkResponse.hostedCheckoutId,
            redirectUrl: sdkResponse.redirectUrl,
            returnMAC: sdkResponse.RETURNMAC,
            amount: undefined,
            currency: undefined,
        };
    },

    fromGetHostedCheckoutSdkResponse(
        sdkResponse: Domain.GetHostedCheckoutResponse,
    ): GetPaymentByHostedCheckoutResponseDto {
        return {
            paymentId: sdkResponse.createdPaymentOutput?.payment?.id,
            status: sdkResponse.status,
        };
    },
};
